core/src/main/scala/org/apache/spark/sql/delta/SnapshotManagement.scala

```diff
@@ -827,7 +827,7 @@ case class LogSegment(
     obj match {
       case other: LogSegment =>
         version == other.version && lastCommitTimestamp == other.lastCommitTimestamp &&
-          logPath == other.logPath
+          logPath == other.logPath && checkpointVersionOpt == other.checkpointVersionOpt
```

Comment:
A dump question: which place replies on this to decide whether to replace a snapshot?

Reply:
What I understand is that when reading or writing, in order to get the latest version of deltalog, `deltalog.update()` will be called first to see if there is already a snapshot and whether it needs to be updated, and finally in the `installLogSegmentInternal()` method to determine whether the snapshot needs to be updated.

Reply:
this is the line the change affects: https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/SnapshotManagement.scala#L580